assets/src/edit-story/components/header/buttons.js

```diff
@@ -277,24 +278,52 @@ function Loading() {
 }
 
 function Buttons() {
-  const { status } = useStory((state) => ({
-    status: state.state.story.status,
-  }));
+  const { status, storyId, link, isFreshlyPublished } = useStory(
+    ({
+      state: {
+        story: { status, storyId, link },
+        meta: { isFreshlyPublished },
+      },
+    }) => ({
+      status,
+      storyId,
+      link,
+      isFreshlyPublished,
+    })
+  );
+  const [showDialog, setShowDialog] = useState(isFreshlyPublished);
+  useEffect(() => {
+    setShowDialog(isFreshlyPublished);
+  }, [isFreshlyPublished]);
+
   const isDraft = 'draft' === status;
+
+  const confirmURL = addQueryArgs('post-new.php', {
+    ['from-web-story']: storyId,
+  });
+
   return (
-    <ButtonList>
-      <List>
-        <Loading />
-        {isDraft && <Update />}
-        {!isDraft && <SwitchToDraft />}
-        <Space />
-        <PreviewButton />
-        <Space />
-        {isDraft && <Publish />}
-        {!isDraft && <Update />}
-        <Space />
-      </List>
-    </ButtonList>
+    <>
+      <ButtonList>
+        <List>
+          <Loading />
+          {isDraft && <Update />}
+          {!isDraft && <SwitchToDraft />}
+          <Space />
+          <PreviewButton />
+          <Space />
+          {isDraft && <Publish />}
+          {!isDraft && <Update />}
+          <Space />
+        </List>
+      </ButtonList>
+      <PostPublishDialog
```

Comment:
Shouldn't it only render this if the dialog is open?

Reply:
That's how react-modal is intended to be used

Reply:
Should it be a conditional render. So wrap this tag in `if(showDialog)`

Reply:
The dialog will already render nothing if it's not open. So there's no benefit in doing that.

See examples at http://reactcommunity.org/react-modal/

Reply:
It's a react _anti-pattern_ (IMO), but it's how this particular component is designed, so let's just stick with that.